### ISSUE TITLE
Uncouple routing from docker-compose files

### ISSUE THREAD

**Issue by sjtrny** [containous/traefik#4553]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

Traefik's routing should be specified in Traefik's configuration, not in docker-compose labels, so that containers can be network and environment agnostic.

Below I outline the current methods for routing with docker, their problems and finally list requirements for a solution.

## Method 1: Labels
Consider the case where one wants to deploy two nginx installations. Their directory structure might be the following

    - traefik/
    	- docker-compose.yml
    	- traefik.toml
    - site1/
    	- docker-compose.yml
    	- public_html/
    - site2/
    	- docker-compose.yml
    	- public_html/

Where ``site1/docker-compose.yml`` and ``site2/docker-compose.yml`` are identical (except for the frontend rule) and of the form:

    version: "3"  
  
	services:  
	  nginx:  
	    image: nginx:latest
	    labels:
	      - traefik.backend=nginx
	      - traefik.frontend.rule=Host:siteX.com
	      - traefik.docker.network=proxy
	      - traefik.port=80
	    ....

but the contents of ``site1/public_html/`` and ``site2/public_html/`` are different.

Currently when one develops ``site1`` they must insert the Traefik labels as above. In other words they are tightly coupling their service to Traefik and must be aware of all other services running on the machine.

## Method 2: Specify front/backend in traefik.toml

The alternative method of specifying frontend and backends in the ``traefik.toml`` file is slightly better as the services do not need to add so many labels.

For example one could set ``site1/docker-compose.yml`` and ``site2/docker-compose.yml`` to:

    version: "3"  
  
	services:  
	  siteX_nginx:  
	    image: nginx:latest

Then in ``traefik.toml`` you can specify routes as with

    [backends]  
      [backends.backend1]  
        [backends.backend1.servers]  
          [backends.backend1.servers.server0]  
          url = "http://site1_nginx"  
      [backends.backend2]  
        [backends.backend2.servers]  
          [backends.backend2.servers.server0]  
          url = "http://site2_nginx"  
      
    [frontends]  
      [frontends.frontend1]  
      backend = "backend1"  
        [frontends.frontend1.routes]  
          [frontends.frontend1.routes.route0]  
          rule = "Host:site1.com"  
      [frontends.frontend2]  
      backend = "backend2"  
        [frontends.frontend2.routes]  
          [frontends.frontend2.routes.route0]  
          rule = "Host:site2.com"

The issue with this configuration is that since the backends require a URL instead of a Host, we have to make sure each service has a unique name.

## Requirements for a Solution

1. Configure routing with zero changes to docker-compose files
2. Allow backends to be specified in the format of "{docker_service}.{docker_project}"

**Comment by geraldcroes** [comment]
Hi!
Thank you for your feedback and having taken the time to explain your use case.

Even though I think I understand your issue, I'm not sure about the proposal and think we need to discuss this further.

For now, I'm tagging the issue and will wait for others to react.

**Comment by decentral1se** [comment]
Yeah, I've ended up while trying to design my container stacks to be to somewhat configurable / agnostic with regard to the proxy. I think method 3. works best for me in my case.

**Comment by emilevauge** [comment]
Hello there,

This proposal has been open for 6 months with minimal traction. We won't be adding it to our roadmap at this time and prefer to close it for now.

If circumstances change or interest grows, we can always reopen this later.
Thank you, @sjtrny, for the suggestion!
